### ISSUE TITLE
Subsequent navigational methods returns empty array

### ISSUE THREAD

**Issue by boredelf** [masvis/angular4-hal#66]
Hello! :)

First, thanks for developing this. It's proving to be very useful to me. Now, the issue: I read the README and other issues and even the source code, but I can't seem to understand why navigational methods (e.g. `next`, `last`, `first`) aren't working even after calling `getAll` or `search`.

Here is the code:
```js
this.proxyService.getAll({ size: 10, sort: [{ path: 'nome', order: 'ASC' }] })
      .subscribe(p1 => {
        console.info(p1);
        this.proxyService.next().subscribe(p2 => console.info(p2));
      });
```

Please, note that:
- The first request occurs;
- `p1` returns an array, as expected;
- The second requests occurs as well;
- But `p2` is empty.
- There are 56 items in my collection (page size 10, as it can be seen above)

I've checked the responses (`p1` and `p2`) from the backend service and they are correct. The problem seems to be happening while deserializing (in this case, `p2`).

Could you be of any assistance? Thanks!

**Comment by sante85** [comment]
in server have you a paginated repository?

**Comment by boredelf** [comment]
Yes. :)

**Comment by boredelf** [comment]
Hello,

Apologies for the delayed feedback. With further debugging, I was able to identify the cause.

On the method `instantiateResourceCollection` of class `ResourceHelper`, it checks if the response has a status code`> = 200` and `<= 207`. That's fine for the first request (`p1`), since it is a` HttpResponseBase` object, but for the second request (`p2`), its type is `object`, meaning it doesn't have the `status` property (confirmed by the debugger), hence creating an empty array.

If you like, I could try to patch it and create a pull request. Let me know. :)

**Comment by sante85** [comment]
please pull request

**Comment by sante85** [comment]
fix in release 7.8.14
